components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/builders/X509CredentialImpl.java

```diff
@@ -43,8 +60,184 @@
+            }
+        }
+
+        if (privateKey == null) {
+            throw new IdentityException("Cannot find the private key for tenant " + tenantDomain);
+        }
+
+        if (signingCert == null) {
+            throw new IdentityException("Cannot find the certificate.");
+        }
+
+        publicKey = signingCert.getPublicKey();
+    }
+
+    /**
+     * Set private key and X509Certificate from the default KeyStore.
+     *
+     * @param keyStoreManager keyStore Manager.
+     * @throws IdentityException Error in retrieving private key and certificate.
+     */
+    private void initializeCredentialForSuperTenantFromDefaultKeyStore(KeyStoreManager keyStoreManager)
+            throws IdentityException {
+
+        try {
+            privateKey = keyStoreManager.getDefaultPrivateKey();
+            signingCert = keyStoreManager.getDefaultPrimaryCertificate();
+        // This Exception is thrown from the KeyStoreManager.
+        } catch (Exception e) {
+            throw new IdentityException("Error retrieving private key and the certificate for tenant " +
+                    MultitenantConstants.SUPER_TENANT_DOMAIN_NAME, e);
+        }
+    }
+
+    /**
+     * Set private key and X509Certificate from the Sign KeyStore which is defined under Security.SAMLSignKeyStore
+     * in carbon.xml.
+     *
+     * @throws IdentityException Error in keystore.
+     */
+    private void initializeCredentialForSuperTenantFromSignKeyStore() throws IdentityException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Initializing Key Data for super tenant using separate sign key store.");
+        }
+
+        try {
+            if (superTenantSignKeyStore == null) {
+                initializeSuperTenantSignKeyStore();
+            }
+
+            String keyAlias = ServerConfiguration.getInstance().getFirstProperty(
+                    SECURITY_SAML_SIGN_KEY_STORE_KEY_ALIAS);
+            char[] keyPassword = ServerConfiguration.getInstance().getFirstProperty(
+                    SECURITY_SAML_SIGN_KEY_STORE_KEY_PASSWORD).toCharArray();
+            Key privateKey = superTenantSignKeyStore.getKey(keyAlias, keyPassword);
+            Certificate publicKey = superTenantSignKeyStore.getCertificate(keyAlias);
+
+            if (privateKey instanceof PrivateKey) {
+                this.privateKey = (PrivateKey) privateKey;
+            } else {
+                throw new IdentityException("Configured signing KeyStore private key is invalid.");
+            }
+
+            if (publicKey instanceof X509Certificate) {
+                this.signingCert = (X509Certificate) publicKey;
+            } else {
+                throw new IdentityException("Configured signing KeyStore X509Certificate is invalid.");
+            }
+        } catch (NoSuchAlgorithmException | UnrecoverableKeyException | KeyStoreException e) {
+            throw new IdentityException("Unable to load keystore", e);
```

Comment:
We can improve the message saying what actually we tried to do.. As per your method name , we are trying to load signing key store for super tenant